### DISCUSSION TITLE
TimeCoordinates Guide

### DISCUSSION THREAD

**Discussion by sonriks6** [google/meridian#302] [Q&A] [answered]
Hello, I don't get the point of adding the TimeCoordinates class as a wrapper for dates, it's suppossed you use a well formated dataset as input but anyway could you update the "Getting Started" notebook to include an example of how to parse the 'time' argument in data loader or explain further how the TimeCoordinates can help in our analysis?

**Comment by santoso-wijaya** [comment, marked as answer]
The `TimeCoordinates` dataclass was introduced to organize how the Meridian parses date labels (strings) in the `time` and `media_time` coordinates in the input data. It defines computed properties that make it easier for analyzer, visualizer, etc modules to deal with dates and date selection logic.

That said, the core model itself treats these date coordinates as simple labels (strings). The `TimeCoordinates` wrapper only comes into play post training.

You should not have to interact directly with the `TimeCoordinates` dataclass. You can use the input loader to load your input data set (which should already have well formed date coordinate labels) as is, and the loader will leverage `TimeCoordinates` internally to validate them.
